source/interface/azure_iot_mqtt.h

```diff
@@ -0,0 +1,380 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file azure_iot_mqtt.h
+ *
+ * @brief Definition for the Azure IoT MQTT.
```

Comment:
What is "Azure IoT MQTT"?  Does this mean the abstraction layer between the core layer and the MQTT stack we'll use?  In any case recommend adding a bit more context to this.